src/shared/internal-calendar/internal-calendar-day.tsx

```diff
@@ -103,12 +108,29 @@ export const InternalCalendarDay = ({
     };
 
     const isDisabled = (day: Dayjs): boolean => {
-        const isOutsideBetweenRange =
-            between && !day.isBetween(between[0], between[1], "day", "[]");
+        // logic to disable dates if it falls outside of either minDate or maxDate
+        if (
+            InternalCalendarHelper.isOutOfDateRange(
+                day,
+                minDate,
+                maxDate,
+                "day"
+            )
+        ) {
+            disabledDates.push(day.format("YYYY-MM-DD"));
```

Comment:
this would mutate the prop. not recommended. i think this function can be reworked to properly short circuit instead.
- true if in disabled dates
- true if failed min max or between check
- true if isStartAfterEnd
- true if isEndBeforeStart
- false otherwise